hooks/useUnmarshalBalance.ts

```diff
@@ -0,0 +1,257 @@
+          }),
+        ]);
+
+        let ethereumTokens: TokenBalance[] = [];
+        let fuseTokens: TokenBalance[] = [];
+        let rate = BigInt(0);
+
+        // Process Ethereum response (Unmarshal)
+        if (ethereumResponse.status === 'fulfilled' && ethereumResponse.value.ok) {
+          const ethereumData: UnmarshalResponse = await ethereumResponse.value.json();
+          // Add chainId to Ethereum tokens
+          ethereumTokens = (ethereumData.assets || []).map(token => ({
+            ...token,
+            chainId: ETHEREUM_CHAIN_ID,
+          }));
+        } else if (ethereumResponse.status === 'rejected') {
+          console.warn('Failed to fetch Ethereum balances:', ethereumResponse.reason);
+        }
+
+        // Process Fuse response (Blockscout)
+        if (fuseResponse.status === 'fulfilled' && fuseResponse.value.ok) {
+          const fuseData: BlockscoutResponse = await fuseResponse.value.json();
+          // Convert Blockscout format to our standard format
+          fuseTokens = fuseData.map((item): TokenBalance => ({
+            contract_ticker_symbol: item.token.symbol,
+            contract_name: item.token.name,
+            contract_address: item.token.address,
+            balance: item.value,
+            quote_rate: item.token.exchange_rate ? parseFloat(item.token.exchange_rate) : 0,
+            quote: 0, // Will be calculated
+            logo_url: item.token.icon_url,
+            contract_decimals: parseInt(item.token.decimals),
+            type: item.token.type,
+            verified: true, // Assume verified if it's on Blockscout
+            chainId: FUSE_CHAIN_ID,
+          }));
+        } else if (fuseResponse.status === 'rejected') {
+          console.warn('Failed to fetch Fuse balances:', fuseResponse.reason);
+        }
+
+        // Process soUSD rate
+        if (soUSDRate.status === 'fulfilled') {
+          rate = soUSDRate.value;
+        } else {
+          console.warn('Failed to fetch soUSD rate:', soUSDRate.reason);
+        }
+
+        // Helper function to check if token is soUSD (vault token)
+        const isSoUSDToken = (token: TokenBalance): boolean => {
+          return token.contract_address.toLowerCase() === ADDRESSES.ethereum.vault.toLowerCase() ||
+                 token.contract_address.toLowerCase() === ADDRESSES.fuse.vault.toLowerCase();
+        };
+
+        // Calculate token values separately for soUSD and regular tokens
+        const calculateTokenValue = (token: TokenBalance): { soUSDValue: number; regularValue: number } => {
+          if (!token.balance) return { soUSDValue: 0, regularValue: 0 };
+
+          // Convert balance from raw format to decimal format using contract_decimals
+          const formattedBalance = parseFloat(token.balance) / Math.pow(10, token.contract_decimals);
+          
+          // Special handling for soUSD tokens - use rate from AccountantWithRateProviders
+          if (isSoUSDToken(token) && rate > 0) {
```

Comment:
@MayankMittal1 Can you please check If the calculation of the `soUSDValue` is correct?